lib/chef/resource.rb

```diff
@@ -138,8 +138,11 @@ def initialize(name, run_context=nil)
     #
     def name(name=nil)
       if !name.nil?
-        raise ArgumentError, "name must be a string!" unless name.kind_of?(String)
-        @name = name
+        if name.is_a?(Array)
+          @name = name.join(', ')
+        else
```

Comment:
I'm sort of agnostic, and I know people are being lazy and passing exception objects into the log resource already (which works because of the prior bypassing in the constructor and then we added coercion into a string in the resource reporter, and it seems like notifies/subscribes already coerces as well).  As long as the setter enforces the coercion and now the setter is used consistently, i'm happy...   But, yeah, that's obviously easy to add...